src/search.cpp

```diff
@@ -157,9 +155,8 @@ void Search::Worker::start_searching() {
     if (rootMoves.empty())
     {
         rootMoves.emplace_back(Move::none());
-        sync_cout << "info depth 0 score "
-                  << UCI::to_score(rootPos.checkers() ? -VALUE_MATE : VALUE_DRAW, rootPos)
-                  << sync_endl;
+        main_manager()->updates.onUpdateShort(
+          {0, UCI::to_score(rootPos.checkers() ? -VALUE_MATE : VALUE_DRAW, rootPos)});
```

Comment:
I'd consider encapsulating the score in a variant or a variant-like class. The conversion to string should be done within UCI layer.

Reply:
and what would be the types of the variant here?

Reply:
some strong types for mate, tbwin, cp

Reply:
Are you thinking of something like this?
https://godbolt.org/z/rn9eYG7Tb

Reply:
yea, the structure is what I had in mind, however I'd rather use visit for the rest as it plays nicely https://godbolt.org/z/nMKa1j6ax. `getValue` makes the encapsulation kinda pointless.

Reply:
Please have a look at https://github.com/official-stockfish/Stockfish/pull/5147/commits/c12fcd7d9e2dbf5fc493cfc847c679220493f244, and check if you like that.